src/resources/js/components/Authentication.vue

```diff
@@ -73,14 +73,22 @@ export default {
           if (response.data.token_type === 'Bearer') {
             const token = response.data.access_token;
             localStorage.setItem('Authorization-token', token); // store the token in localstorage
+            // this window.axios code may be deleted when login forces a hard refresh
+            window.axios.defaults.headers.common = {
+              Authorization: `Bearer ${localStorage.getItem('Authorization-token')}`, // put it directly in header
+            };
+            this.$router.go(0); // this is a disgusting hack, it forces a refresh, and as
+            // our "current page" is the root, well it goes back to it
+            // (the correct way would have been to update the value in usertooltip
+            // through one of many way vuejs let us do that)
+            this.$router.push({ name: 'Hello' }); // all routing is handled by vuejs, should be changed for the final home route
             alert.showMessage('success', 'Authentifié');
-            this.$router.replace({ name: 'Hello' }); // all routing is handled by vuejs, should be changed for the final home route
           } else {
             alert.showMessage('error', 'Réponse du serveur inatendue');
           }
```

Comment:
it's the exact same code as AuccountCreation should be merged into a function that will be used by both components

Reply:
putting it all in a function in app.js has an extremely weird behaviour, it will first go to .catch, then do the .then, and while doing the then, it will do things from the function out of ordre which remind me of all the problems I got with the first edition of this code

Reply:
I also tried to put it all in a function in app.vue, but couldnt access it with this.$parent.functionIMade();